### PR TITLE
added first version of chemical formula parsing , incl. unit tests

### PR DESCRIPTION
**Pull request by michaelwitting**
This PR contains a prototype version of the chemical formula parsing

### CODE REVIEW

File: R/chemFormula.R [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+#' @import stringr str_extract_all str_extract
+#'
+#' @export
+#'
+#' @examples
+#' 
+#' elements <- c("C" = 6, "H" = 12, "O" = 6)
+#' list2formula(elements)
+list2formula <- function(x) {
+
```
Comment (jorainer) [inline R/chemFormula.R:67]: maybe add a check if `x` does not have names. Something like `if (is.null(names(x))) stop("'x' should be a named vector")`.

**Comment by michaelwitting** [comment]
What about countElements and pasteElements?Von meinem Samsung Galaxy Smartphone gesendet.
-------- Ursprüngliche Nachricht --------Von: Johannes Rainer <notifications@github.com> Datum: 10.04.20  08:16  (GMT+01:00) An: rformassspectrometry/MetaboCoreUtils <MetaboCoreUtils@noreply.github.com> Cc: Michael Witting <michael.witting@helmholtz-muenchen.de>, Mention <mention@noreply.github.com> Betreff: Re: [rformassspectrometry/MetaboCoreUtils] added first version of chemical formula parsing , incl. unit tests (#13) 
@jorainer requested changes on this pull request.

Thanks @michaelwitting for the PR! I have some minor suggestions. In addition I am not too happy with the name of the functions formula2list and list2formula, because the functions actually don't return a list. We should find better names. formula2list could be elementCount (because in fact it returns the count of elements). I'm just not sure about a better name for the reverse operation, maybe pasteElements? Happy for alternative suggestions.

In R/chemFormula.R:
> @@ -0,0 +1,127 @@
+#' @title Parse a chemical formula into a named vector
+#'
+#' @description
+#'
+#' `formula2list` This function parses a chemical into a named vector
+#'
+#' @param x `character` Single string with chemical
+#'
+#' @return `numeric` A named list containing the parsed chemical formula
+#'
+#' @author Michael Witting
+#' 
+#' @import stringr str_extract_all str_extract

I think this should be @importFrom stringr str_extract_all str_extract



In R/chemFormula.R:
> +#' @param x `character` A named vector containing the number of elements
+#'
+#' @return `character` Single string with the chemical formula
+#'
+#' @author Michael Witting
+#' 
+#' @import stringr str_extract_all str_extract
+#'
+#' @export
+#'
+#' @examples
+#' 
+#' elements <- c("C" = 6, "H" = 12, "O" = 6)
+#' list2formula(elements)
+list2formula <- function(x) {
+  

maybe add a check if x does not have names. Something like if (is.null(names(x))) stop("'x' should be a named vector").

—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or unsubscribe.
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/rformassspectrometry/MetaboCoreUtils/pull/13#pullrequestreview-391279572",
"url": "https://github.com/rformassspectrometry/MetaboCoreUtils/pull/13#pullrequestreview-391279572",
"name": "View Pull Request"
},
"description": "View this Pull Request on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]
Helmholtz Zentrum Münche
Helmholtz Zentrum Muenchen
Deutsches Forschungszentrum fuer Gesundheit und Umwelt (GmbH)
Ingolstaedter Landstr. 1
85764 Neuherberg
www.helmholtz-muenchen.de
Aufsichtsratsvorsitzende: MinDir.in Prof. Dr. Veronika von Messling
Geschaeftsfuehrung: Prof. Dr. med. Dr. h.c. Matthias Tschoep, Kerstin Guenther
Registergericht: Amtsgericht Muenchen HRB 6466
USt-IdNr: DE 129521671

File: R/chemFormula.R [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+#'
+#' `formula2list` This function parses a chemical into a named vector
+#'
+#' @param x `character` Single string with chemical
+#'
+#' @return `numeric` A named list containing the parsed chemical formula
+#'
+#' @author Michael Witting
+#' 
+#' @import stringr str_extract_all str_extract
```
[the remark this answers is not in the document]
Comment (michaelwitting) [reply R/chemFormula.R:13]: Oopss... Yes you are right

**Comment by jorainer** [comment]
Yes, I like `countElements` and `pasteElements`! Good suggestion.

**Pull request merged**